src/atsu/src/com/HoppieConnector.ts

```diff
@@ -0,0 +1,149 @@
+    }
+
+    public static async poll(): Promise<[AtsuStatusCodes, AtsuMessage[]]> {
+        const retval: AtsuMessage[] = [];
+
+        if (SimVar.GetSimVarValue('L:A32NX_HOPPIE_ACTIVE', 'number') !== 1) {
+            return [AtsuStatusCodes.NoHoppieConnection, retval];
+        }
+
+        const flightNo = SimVar.GetSimVarValue('ATC FLIGHT NUMBER', 'string');
+        if (flightNo.length === 0 || flightNo === '1123') {
+            return [AtsuStatusCodes.ComFailed, retval];
+        }
+
+        const text = await NXApi.sendHoppieRequest(NXDataStore.get('CONFIG_HOPPIE_USERID', ''), flightNo, flightNo, 'poll', '');
+
+        // something went wrong
+        if (!text.startsWith('ok')) {
+            return [AtsuStatusCodes.ComFailed, retval];
+        }
+
+        // split up the received data into multiple messages
+        let messages = text.split(/({.*?})/gm);
+        messages = messages.filter((elem) => elem !== 'ok' && elem !== 'ok ' && elem !== '} ' && elem !== '}' && elem !== '');
+
+        // create the messages
+        messages.forEach((element) => {
+            // get the single entries of the message
+            // example: [CALLSIGN telex, {Hello world!}]
+            const entries = element.substring(1).split(/({.*?})/gm);
+
+            // get all relevant information
+            const metadata = entries[0].split(' ');
+            const sender = metadata[0].toUpperCase();
+            const type = metadata[1].toLowerCase();
+            const content = entries[1].replace(/{/, '').replace(/}/, '');
+
+            switch (type) {
+            case 'telex':
+                const freetext = new FreetextMessage();
+                freetext.Network = AtsuMessageNetwork.Hoppie;
+                freetext.Station = sender;
+                freetext.Direction = AtsuMessageDirection.Input;
+                freetext.ComStatus = AtsuMessageComStatus.Received;
+                freetext.Message = content.replace(/\n/i, ' ');
+                retval.push(freetext);
+                break;
+            case 'cpdlc':
+                const cpdlc = new CpdlcMessage();
+                cpdlc.Station = sender;
+                cpdlc.Direction = AtsuMessageDirection.Input;
+                cpdlc.ComStatus = AtsuMessageComStatus.Received;
+
+                // split up the data
+                const elements = content.split('/');
+                cpdlc.CurrentTransmissionId = parseInt(elements[2]);
+                if (elements[3] !== '') {
+                    cpdlc.PreviousTransmissionId = parseInt(elements[3]);
+                }
+                cpdlc.RequestedResponses = stringToCpdlc(elements[4]);
+                cpdlc.Message = elements[5];
+
+                retval.push(cpdlc);
+                break;
+            default:
+                break;
+            }
+        });
+
+        return [AtsuStatusCodes.Ok, retval];
+    }
+
+    public static pollInterval(): number {
+        return 1000;
```

Comment:
Yes, this can also work. The IRL simulation is done in Datalink.js.
This queues (virtually) all send and receive requests and handles them step by step.
So the IRL simulation is not part of the poll-function

Reply:
updated to 5000